structuralcodes/sections/specific.py

```diff
@@ -0,0 +1,446 @@
+
+class RectangularSection(GenericSection): 
+    """This is the specific implementation of a rectangular class section.
+
+    The section is a 2D geometry where Y axis is horizontal while Z axis is
+    vertical.
+
+    The moments and curvatures around Y and Z axes are assumed positive
+    according to RHR.
+
+    Attributes: TODO
+        geometry (Union(SurfaceGeometry, CompoundGeometry)): The geometry of
+            the section.
+        name (str): The name of the section.
+        section_calculator (GenericSectionCalculator): The object responsible
+            for performing different calculations on the section (e.g. bending
+            strength, moment curvature, etc.).
+    """
+
+    def __init__(self, height: float, width: float, concrete: Concrete, name: t.Optional[str] = None, integrator: t.Literal['marin', 'fiber'] = 'marin') -> None:
+        """Initialize a GenericSection.
+
+        Arguments:
+            geometry (Union(SurfaceGeometry, CompoundGeometry)): The geometry
+                of the section.
+            name (str): The name of the section.
+            integrator (str): The name of the SectionIntegrator to use.
+        """
+        poly: Polygon = Polygon(((0, 0), (width, 0), (width, height), (0, height)))
+        geometry: SurfaceGeometry = SurfaceGeometry(poly, concrete)
+        geometry.translate(-width/2, -height/2)
+        if name is None:
+            name: str = 'RectangularSection'
+        super().__init__(geometry, name, integrator)
+
+        self.height: float = height
+        self.width: float = width
+
+        # Calculating gross section properties
+        # Gross area
+        area: float = height * width
+        perimeter: float = 2 * (height + width)
+        ea: float = area * geometry.material.get_tangent(eps=0)[0]
+        for geo in self.geometry.geometries:
+            pass
+        for geo in self.geometry.point_geometries:
+            pass
+
+        if geometry.density is not None:
+            mass = area * geometry.density * 1e-9
+        else:
+            mass = 0
+
+
+        gp = s_res.GrossProperties()
+        gp.area = area
+        gp.perimeter = perimeter
+        gp.ea = ea
+        gp.mass = mass
+        # TODO: more gross sectional properties, and store them in self._gross_properties
+
+    @property
+    def area(self) -> float:
+        """Returns Area of the cross section.
+        Only the gross area of concrete is computed.
+
+        Returns:
+            float: The area of concrete section.
+        """
+        return self.width * self.height
+
+
+class RectangularRC(Section):
+    """Class for RC rectangular cross section.
+    The class assumes width and height respectively
+    along local x and y axes.
+
+    Input parameters:
+    width: width of the section
+    height: height of the section
+    cover: net cover (distance from outer edget of the section to
+    edge of the stirrup)
+    reinforcement_data: ReinforcementData
```

Comment:
This seems not defined?

Reply:
Not relevant (this is your old code which I have pasted in as a help)

Reply:
Please try to keep the PR clean. You can have a local copy of the code I shared, but in the PR I suggest we leave only the code relevant for the PR itself and that eventually (at the end of the review process) will be merged with dev.